homeassistant/components/thethingsnetwork/__init__.py

```diff
@@ -33,10 +27,65 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Initialize of The Things Network component."""
-    conf = config[DOMAIN]
-    app_id = conf.get(CONF_APP_ID)
-    access_key = conf.get(CONF_ACCESS_KEY)
 
-    hass.data[DATA_TTN] = {TTN_ACCESS_KEY: access_key, TTN_APP_ID: app_id}
+    if DOMAIN in config:
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            "manual_migration",
+            breaks_in_ha_version="2021.10.0",
+            is_fixable=False,
+            severity=ir.IssueSeverity.ERROR,
+            translation_key="manual_migration",
+            translation_placeholders={"domain": DOMAIN},
+        )
 
     return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Establish connection with The Things Network."""
+
+    _LOGGER.debug(
+        "Set up %s at %s",
+        entry.data[CONF_APP_ID],
+        entry.data.get(CONF_HOSTNAME, TTN_API_HOSTNAME),
+    )
```

Comment:
I think it would be nicer to implement diagnostics in a follow up to get debug information

Reply:
good idea - should I remove the debug logs for now or keep them for now and replace them with diagnostics in a follow up next PR?

Reply:
added task for this at https://github.com/home-assistant/core/issues/112491